src/cowboy_http2.erl

```diff
@@ -238,8 +245,12 @@ loop(State=#state{parent=Parent, socket=Socket, transport=Transport,
 			setopts_active(State),
 			loop(State, Buffer);
 		%% System messages.
+		{'EXIT', Parent, shutdown} ->
+			%% Our supervisor attemts to shut us down and expects us
+			%% to terminate with reason 'shutdown' or 'normal'.
+			loop(graceful_shutdown(State, shutdown), Buffer);
```

Comment:
There doesn't seem to be any need to provide the Reason here, and in timeout messages, it's always `shutdown`.

Reply:
`sys:terminate/2,3` expects the process to exit with the same Reason as provided to `sys:terminate/2,3` and a supervisor expects `shutdown` in particular, but `cowboy_http2` exit Reasons always seem to be wrapped in extra tuples for debugging, so this doesn't work anyway. If there is no plan to change this (i.e. to exit with the requested reason), it's better to remove the Reason parameter here, sure.

Reply:
Right, nevermind, I missed the `sys:terminate` stuff.